compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -367,7 +367,9 @@ class Typer extends Namer
             if (qualifies(defDenot)) {
               val found =
                 if (isSelfDenot(defDenot)) curOwner.enclosingClass.thisType
-                else {
+                else if(ctx.isJava && defDenot.symbol.isStatic) {
```

Comment:
one last controversial `if(` spacing to fix with quiet syntax. Oh, just realized that it's called "quiet" syntax because no one is arguing about it.